pootle/static/js/editor/components/SuggestionFeedbackForm.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) Pootle contributors.
+ *
+ * This file is a part of the Pootle project. It is distributed under the GPL3
+ * or later license. See the LICENSE file for a copy of the license and the
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import React from 'react';
+
+import FormElement from 'components/FormElement';
+import FormMixin from 'mixins/FormMixin';
+
+import { decodeEntities } from '../utils';
+
+export const SuggestionFeedBackForm = React.createClass({
+
+  propTypes: {
+    suggId: React.PropTypes.number.isRequired,
+    suggText: React.PropTypes.string.isRequired,
+    editor: React.PropTypes.object.isRequired,
+  },
+
+  mixins: [FormMixin],
+
+  /* Lifecycle */
+
+  getInitialState() {
+    return {
+      formData: {
+        comment: '',
+        translation: this.props.suggText,
+      },
+    };
+  },
+
+  fields: ['comment'],
+
+  /* Handlers */
+
+  handleAccept(e) {
+    e.preventDefault();
+    if (this.state.formData.translation === this.props.suggText) {
+      this.props.editor.acceptSuggestion(this.props.suggId, this.state.formData);
+    } else {
+      const area = document.querySelector('.js-translation-area');
+      area.value = decodeEntities(this.state.formData.translation);
```

Comment:
This is not the component's responsibility, and it must be blind to how the outside world is and/or works, i.e it must not add coupling. So avoid doing this at all costs: these type of things must be handled by callbacks passed via props from higher levels in the hierarchy.